cmd/thanos/tools.go

```diff
@@ -34,31 +36,42 @@ func registerCheckRules(app extkingpin.AppClause) {
 	})
 }
 
-func checkRulesFiles(logger log.Logger, files *[]string) error {
+func checkRulesFiles(logger log.Logger, patterns *[]string) error {
 	var failed errutil.MultiError
 
-	for _, fn := range *files {
-		level.Info(logger).Log("msg", "checking", "filename", fn)
-		f, err := os.Open(fn)
-		if err != nil {
+	for _, p := range *patterns {
+		level.Info(logger).Log("msg", "checking", "pattern", p)
+		matches, err := filepath.Glob(p)
+		if err != nil || matches == nil {
+			err = errors.New("matching file not found")
 			level.Error(logger).Log("result", "FAILED", "error", err)
 			level.Info(logger).Log()
 			failed.Add(err)
 			continue
 		}
-		defer func() { _ = f.Close() }()
+		for _, fn := range matches {
+			level.Info(logger).Log("msg", "checking", "filename", fn)
+			f, er := os.Open(fn)
+			if er != nil {
+				level.Error(logger).Log("result", "FAILED", "error", er)
+				level.Info(logger).Log()
+				failed.Add(err)
+				continue
+			}
+			defer func() { _ = f.Close() }()
```

Comment:
Typically we'd use `rununtil.CloseWithLogOnErr` but since at the end the process closes immediately, this is fine. Just something to keep in mind for future contributions :+1:

Reply:
Got it, Thanks :raised_hands: